File: Civitas - Java/src/civitas/CasillaCalle.java

package civitas;

import java.util.ArrayList;


public class CasillaCalle extends Casilla {
    private TituloPropiedad titulo;
    
    //Tipo CALLE
    CasillaCalle(TituloPropiedad titulo){
        super( titulo.getNombre() );
        this.tituloPropiedad = titulo;
        this.nombre = titulo.getNombre();
        this.importe = titulo.getPrecioCompra();
    }
    
    //Si el jugador existe se genera un informe, y se le permite comprar si no tiene propietario o paga el aquiler si lo hay.
    @Override
    protected void recibeJugador(int iactual, ArrayList<Jugador> todos){
        Jugador jugador;
        if( jugadorCorrecto(iactual, todos) ){
            //1:
            informe(iactual, todos);
            //2:
            jugador = todos.get(iactual);
            
            //alt 3:
            if( !tituloPropiedad.tienePropietario() )
                jugador.puedeComprarCasilla();
            else{
                //4:
                tituloPropiedad.tramitarAlquiler(jugador);
            }
        }
    }
      
    //Getter de TituloPropiedad
    TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
    
    @Override
    public String toString(){
        return tituloPropiedad.toString();
    }
    
}

File: Civitas - Ruby/lib/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require './Titulo_propiedad'

module Civitas
  class Casilla_calle < Casilla
    attr_reader :titulo_propiedad
    
    def initialize titulo
      super(titulo.nombre, titulo.precio_compra )
      @titulo_propiedad = titulo
      @importe = titulo.precio_compra
    end
    
    #Jugador cae en calle
    def recibe_jugador(iactual, todos)
      if( jugador_correcto(iactual, todos) )
          informe(iactual, todos)
          jugador = todos[iactual]
          if( !@titulo_propiedad.tiene_propietario )
              jugador.puede_comprar_casilla
          else
              @titulo_propiedad.tramitar_alquiler(jugador)    
          end
      end
    end
    
    def to_s
      return @titulo_propiedad.to_s
    end
    
  end
end
